### ISSUE TITLE
Varible name suggested same as foreach varible name

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#31304]
In 15.8.7
Using the following code makes Visual Studio suggest I use the name item for class Item, even though that name is already defined.

Note that var is NOT an Item class.

This is obviously incorrect as there will be two different varibles with the same item name.

```csharp
foreach (var item in returnedShipment.items)
{
  shipment.itemReferences.Add(item.item_reference);
  Item <name item is suggested here>
}
```

![devenv_2018-10-18_11-01-46 2](https://user-images.githubusercontent.com/12449387/48872550-69c60180-ed9e-11e8-850f-16f28f1c6eef.png)

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/361110/varible-name-suggested-same-as-foreach-varible-nam.html
VSTS ticketId: 708453_
_These are the original issue comments:_
(no comments)
_These are the original issue solutions:_
(no solutions)

**Comment by dibarbet** [comment]
This is wider than just foreach, for example

```csharp
class ClassA { }
class ClassB { }

void M()
{
    var classB = new ClassA();
    ClassB ^
}
```
If you're at the caret position, it will suggest classB as the name, whatever the type may be of the original classB variable.  We must not be looking to see if the variable already exists when providing the suggestion.

<img width="253" alt="namesuggestion" src="https://user-images.githubusercontent.com/5749229/51407540-0394fb00-1b11-11e9-9b0e-6af4f066a83e.png">

Currently investigating the best way of checking the symbols existence.

**Comment by dibarbet** [comment]
Resolved for 16.1 prev 1 in https://github.com/dotnet/roslyn/commit/2fdc2508cd9b3c5f90f1973be6ddaf900ee21c5c
